aries_cloudagent/anoncreds/routes.py

```diff
@@ -56,91 +59,80 @@
 
 SPEC_URI = "https://hyperledger.github.io/anoncreds-spec"
 
+endorser_connection_id_description = """
+    Connection identifier (optional) (this is an example)
+    You can set this is you know the endorsers connection id you want to use.
+    If not specified then the agent will attempt to find an endorser connection."""
+create_transaction_for_endorser_description = """
+    Create transaction for endorser (optional, default false). 
+    Use this for agents who don't specify an author role but want to 
+    create a transaction for an endorser to sign."""
+
 
 class SchemaIdMatchInfo(OpenAPISchema):
     """Path parameters and validators for request taking schema id."""
 
     schema_id = fields.Str(
-        description="Schema identifier",
-        example=INDY_SCHEMA_ID_EXAMPLE,
+        metadata={
+            "description": "Schema identifier",
+            "example": INDY_SCHEMA_ID_EXAMPLE,
+        }
     )
 
 
-class CredIdMatchInfo(OpenAPISchema):
-    """Path parameters and validators for request taking credential id."""
+class SchemaPostOptionSchema(OpenAPISchema):
+    """Parameters and validators for schema options."""
 
-    cred_def_id = fields.Str(
-        description="Credential identifier",
-        required=True,
-        example=INDY_CRED_DEF_ID_EXAMPLE,
+    endorser_connection_id = fields.Str(
+        metadata={
+            "description": endorser_connection_id_description,
+            "required": False,
```

Comment:
`required` is not a metadata field -- should be passed as `required = False, metadata=...`
I'll fix in my deprecation warnings PR (which had merge conflicts here)